lighty-modules/lighty-jetty-server/src/main/java/io/lighty/server/HttpsLightyServerBuilder.java

```diff
@@ -15,36 +16,36 @@
 import org.eclipse.jetty.server.SecureRequestCustomizer;
 import org.eclipse.jetty.server.Server;
 import org.eclipse.jetty.server.ServerConnector;
-import org.eclipse.jetty.server.SslConnectionFactory;
 
 public class HttpsLightyServerBuilder extends LightyServerBuilder {
-    private final SecurityConfig securityConfig;
+    private SecurityConfig securityConfig;
 
-    public HttpsLightyServerBuilder(final InetSocketAddress inetSocketAddress, final SecurityConfig securityConfig) {
+    public HttpsLightyServerBuilder(final InetSocketAddress inetSocketAddress) {
         super(inetSocketAddress);
-        this.server = new Server();
-        this.securityConfig = securityConfig;
+    }
+
+    public LightyServerBuilder addSecurityConfig(final SecurityConfig config) {
+        this.securityConfig = config;
+        return this;
     }
 
     @Override
     public Server build() {
-        final Server server = super.build();
-        final SslConnectionFactory ssl = securityConfig.getSslConnectionFactory(HttpVersion.HTTP_1_1.asString());
-        final ServerConnector sslConnector = new ServerConnector(server,
-                ssl, httpConfiguration(this.inetSocketAddress));
+        Preconditions.checkState(securityConfig != null);
```

Comment:
How we can enable unsecure connection now? When security config is mandatory?

Reply:
Probably it was better to set security config in constructor if it is required..